sbin/spark-daemon.sh

```diff
@@ -162,6 +162,15 @@ run_command() {
   esac
 
   echo "$newpid" > "$pid"
+
+  for i in {1..10}
+  do
+    if [[ $(ps -p "$newpid" -o comm=) =~ "java" ]]; then
+       break
+    fi
+    sleep 0.5
```

Comment:
OK, that's fair, we still need to see if it dies. I suppose there could be a second loop to keep watching for it to die, but in the normal case that still requires waiting some number of seconds. This is OK, except for the PR problem of course.